R/fct_SQLite.R

```diff
@@ -91,17 +91,99 @@ db_create <- function(
       status = status
     )
   
-  new_data <- list(
+  new_pk_data <- list(
     "all_review_data" = df,
-    "query_data"      = query_data_skeleton,
-    "db_synch_time"   = data.frame(synch_time = data_synch_time)
+    "query_data"      = query_data_skeleton
+  )
+  idx_pk_cols <- list(
+    all_review_data = idx_cols
+  )
+  other_data <- list(
+    "db_synch_time"   = data.frame(synch_time = data_synch_time),
+    "db_version" = data.frame(version = db_version)
   )
   con <- get_db_connection(db_path)
-  for(i in names(new_data)){
+  db_add_tables(con, new_pk_data, idx_pk_cols, other_data)
+  cat("Finished writing to database\n\n")
+}
+
+#' Add new tables to DB
+#'
+#' @param con A DBI Connection to the SQLite DB
+#' @param pk_data A named list of data frames to add a primary key field to DB
+#'   table. Names will correspond to the DB table names.
+#' @param idx_cols A named list of the fields defining unique records for a
+#'   table. Names will correspond to the table to apply the index constraint.
+#' @param other_data A named list of other data frames to add to the DB. Names
+#'   will correspond to the DB table names.
+#'
+#' @keywords internal
+db_add_tables <- function(con, pk_data, idx_cols, other_data) {
```

Comment:
The argument `idx_cols` is different here from the internal object named `idx_cols` (=character vector). Is the correct one used in the function?

I would suggest to use different names, this is confusing for me at least.

Reply:
Done.